bin/plugin/commands/translations.js

```diff
@@ -0,0 +1,168 @@
+ *
+ * @property {string=} directory Optional directory, default is the root `/modules` directory.
+ * @property {string=} output    Optional output PHP file, default is the root `/module-i18n.php` file.
+ */
+
+/**
+ * @typedef WPTranslationsSettings
+ *
+ * @property {string} textDomain Plugin text domain.
+ * @property {string} directory  Modules directory.
+ * @property {string} output     Output PHP file.
+ */
+
+const options = [
+	{
+		argname: '-d, --directory <directory>',
+		description: 'Modules directory',
+	},
+	{
+		argname: '-d, --output <output>',
+		description: 'Output file',
+	},
+];
+
+/**
+ * Command that generates a PHP file from module header translation strings.
+ *
+ * @param {WPTranslationsCommandOptions} opt
+ */
+async function handler( opt ) {
+	await createTranslations( {
+		textDomain: config.textDomain,
+		directory: opt.directory || 'modules',
+		output: opt.output || 'module-i18n.php',
+	} );
+}
+
+module.exports = {
+	options,
+	handler,
+};
+
+const TAB = '\t';
+const NEWLINE = '\n';
+const FILE_HEADER = `<?php
+/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */
+$generated_i18n_strings = array(
+`;
+const FILE_FOOTER = `
+);
+/* THIS IS THE END OF THE GENERATED FILE */
+`;
+
+/**
+ * Parses module header translation strings.
+ *
+ * @param {WPTranslationsSettings} settings Translations settings.
+ *
+ * @return {[]string} List of translation strings.
+ */
+async function getTranslations( settings ) {
+	const moduleFilePattern = path.join( settings.directory, '*/load.php' );
+	const moduleFiles = await glob( path.resolve( '.', moduleFilePattern ) );
+
+	const moduleTranslations = await Promise.all(
+		moduleFiles.map( async ( moduleFile ) => {
+			const fileStream = fs.createReadStream( moduleFile );
+
+			const rl = readline.createInterface( {
+				input: fileStream,
+			} );
+
+			const headers = [ 'Module Name', 'Description' ];
+			const translationStrings = [];
+
+			for await ( const line of rl ) {
+				headers.forEach( ( header, index ) => {
+					const regex = new RegExp(
+						'^(?:[ \t]*<?php)?[ \t/*#@]*' + header + ':(.*)$',
+						'i'
+					);
+					const match = regex.exec( line );
+
+					if ( match ) {
+						const headerValue = match[ 1 ]
+							.replace( /\s*(?:\*\/|\?>).*/, '' )
+							.trim();
+						if ( headerValue ) {
+							headers.splice( index, 1 );
+							translationStrings.push( headerValue );
+						}
+					}
+				} );
+			}
+
+			return translationStrings;
+		} )
+	);
+
+	return moduleTranslations.flat();
+}
+
+/**
+ * Parses module header translation strings.
+ *
+ * @param {[]string}               translations List of translation strings.
+ * @param {WPTranslationsSettings} settings     Translations settings.
+ */
+function createTranslationsPHPFile( translations, settings ) {
+	const output = translations.map( ( translation ) => {
+		// Escape single quotes.
+		return (
+			TAB +
+			`__( '${ translation.replace( /'/g, "\\'" ) }', '${
+				settings.textDomain
+			}' ),`
```

Comment:
Perhaps would be better if we use `_x` to be able to provide an additional context for translators?

```js
_x( '${ translation.replace( /'/g, "\\'" ) }', 'Module name', '${ settings.textDomain }' ),`
```
or
```js
_x( '${ translation.replace( /'/g, "\\'" ) }', 'Module description', '${ settings.textDomain }' ),`
```

Reply:
Great idea! I've refactored this to include a context and use `_x` throughout.